src/obfl-specification.html

```diff
@@ -1557,18 +1562,121 @@ <h4 id="L735" class="include">table-of-contents</h4>
+          </dl></dd>
+        <dt>keep [optional]</dt>
+          <dd>Keep rows in this block together by breaking pages early.</dd>
+          <dd>One of:
+          <dl><dt>'auto' (default)</dt>
+          	<dt>'page'</dt>
+          	<dd>break page early if the whole block doesn't fit on the page.</dd>
+          	<dt>'sheet'</dt>
+          	<dd>break page and sheet early if the whole block doesn't fit on the sheet.</dd>
+          	<dt>'volume'</dt>
+          	<dd>break page, sheet and volume early if the whole block doesn't fit in the volume.</dd>
+           </dl></dd>
+        <dt>keep-with-next [optional]</dt>
+          <dd>Keep the following block's first line(s) on the same page as this
+            block.</dd>
+          <dd>An integer in the range [0, 9].</dd>
+        <dt>orphans [optional]</dt>
+          <dd>Specifies the minimum number of lines of a paragraph that must be
+            left at the bottom of a page.</dd>
+          <dd>An integer in the range [1, 9].</dd>
+          <dd>Default is 1.</dd>
+        <dt>vertical-align [optional]</dt>
+          <dd>One of 'before', 'center', 'after'</dd>
+          <dd>Note that all block properties that effect the height of the
+            block are taken into account when aligning, both visible (i.e.
+            borders) and invisible (i.e. paddings and margins).</dd>
+        <dt>vertical-position [optional]</dt>
+          <dd>Specifies a vertical position for the block in row heights from
+            the top of the page body area.</dd>
+          <dd>Can be relative (percent) or absolute (a positive integer).</dd>
+          <dd>For example: 50% or 12</dd>
+          <dd>Note that vertical-positioning behave like leaders. In other
+            words, the vertical-position attribute can only increase the amount
+            of space preceding a block. It does not rearrange blocks. If a
+            second block has a vertical position lower than the block preceding
+            it, the second blocks vertical-position will either be ignored or
+            rendered on the following page, depending on the specific
+            situation. </dd>
+        <dt>widows [optional]</dt>
+          <dd>Specifies the minimum number of lines of a paragraph that must be
+            left at the top of a page.</dd>
+          <dd>An integer in the range [1, 9].</dd>
+          <dd>Default is 1.</dd>
+      </dl>
+    </dd>
+  <dt>Content Model</dt>
+    <dd>toc-entry, toc-block, text(), leader, marker, br, evaluate, page-number, span, style, anchor, block, table, xml-data [0 or more]</dd>
 </dl>
 </div>
 
 <h4 id="L1105" class="include">toc-entry</h4>
 
-<p>The toc-entry element defines an entry in a table of contents.</p>
+<p>The toc-entry element defines an entry in a table of contents. It is a wrapper
+around inline content. In a volume-ranged toc-sequence, the inline content will
```

Comment:
Hmm, the fact that the element contains inline content is not so relevant for its functionality I think. Besides, you already mention it in the next paragraph. Maybe something like this?

> The toc-entry element defines an entry in a table of contents. It is a conditional container of content. In a volume-ranged toc-sequence, the content will only be rendered if the block the toc-entry refers to starts in the current volume. In a document-ranged toc-sequence, the content will always be rendered.

We also need to say something about what happens with `toc-block` elements if they have no descendant `toc-entry`s that are rendered, because in this case the blocks may not be rendered. But maybe that belongs to the `toc-block` section?